apps/mobile/src/hooks/useRegisterForNotifications.ts

```diff
@@ -0,0 +1,272 @@
+}
+
+interface NotificationsProps {
+  registerForNotifications: () => Promise<RegisterForNotificationsProps>
+  unregisterForNotifications: () => Promise<RegisterForNotificationsProps>
+  updatePermissionsForNotifications: () => Promise<RegisterForNotificationsProps>
+  isLoading: boolean
+  error: string | null
+}
+
+const useRegisterForNotifications = (): NotificationsProps => {
+  // Local states
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+  // Custom hooks
+  const { data: nonceData } = useAuthGetNonceV1Query()
+  const { registerForNotificationsOnBackEnd, unregisterForNotificationsOnBackEnd } = useGTW()
+  const { getNotificationRegisterPayload } = useNotificationPayload()
+  // Redux
+  const dispatch = useAppDispatch()
+  const activeSafe = useDefinedActiveSafe()
+  const delegatedAccounts = useAppSelector(selectDelegatedAccounts)
+  const { ownerFound, accountType } = useNotificationGTWPermissions().getAccountType()
+  /*
+   * Push notifications can be enabled by an two type of users. The owner of the safe or an observer of the safe
+   * In the first case, the owner can subscribe to ALL NotificationTypes listed in @safe-global/store/gateway/AUTO_GENERATED/notifications
+   * including confirmation requests notifications
+   * In the second case, the observer notifications will not include confirmation requests
+   *
+   * We only notify required confirmation events to owners or delegates
+   * to prevent other subscribers from receiving "private" events
+   */
+  const registerForNotifications = useCallback(async () => {
+    try {
+      setLoading(true)
+      setError(null)
+      // Step 1 - Set up Firebase Cloud Messaging steps
+      await FCMService.registerAppWithFCM()
+      await FCMService.saveFCMToken()
+      const fcmToken = await FCMService.getFCMToken()
+      FCMService.listenForMessagesBackground()
+
+      /* Step 3 - Create a new random (delegated) private key to avoid exposing the subscriber's private key
+       *
+       * This key will be used to register for notifications on the backend
+       * avoiding the prompt to grant notifications permission
+       */
+      const randomDelegatedAccount = Wallet.createRandom()
+
+      if (!randomDelegatedAccount) {
+        setLoading(false)
+        setError(ERROR_MSG)
+        return {
+          loading,
+          error,
+        }
+      }
+
+      const accountDetails = {
+        address: randomDelegatedAccount.address,
+        privateKey: randomDelegatedAccount.privateKey, // This is the private key that will be used to sign the message for notification registration only
+        type: accountType,
+      }
+
+      // Step 5 - Store the random delegated account in the Redux store
+      dispatch(
+        addOrUpdateDelegatedAccount({
+          accountDetails,
+          safes: [activeSafe],
+        }),
+      )
+
+      const signer = getSigner(randomDelegatedAccount.privateKey)
```

Comment:
isn't wallet.createRandom equal to new Wallet(PK)? So the random Account is the signer?